packages/core/src/Stores/client-store.js

```diff
@@ -433,12 +434,20 @@ export default class ClientStore extends BaseStore {
 
         reaction(
             () => [this.account_settings],
-            () => {
-                const language = getRedirectionLanguage(this.account_settings?.preferred_language);
+            async () => {
+                const language = getRedirectionLanguage(this.account_settings?.preferred_language, this.is_new_session);
                 window.history.replaceState({}, document.title, urlForLanguage(language));
 
-                this.setPreferredLanguage(language);
-                LocalStore.set(LANGUAGE_KEY, language);
+                if (
+                    language !== this.account_settings?.preferred_language &&
```

Comment:
Can we move this condition up and store value in variable then we use that variable for condition ? 😬

That would be little more readable.